benchmarking/nursery/visualize_plotly/interactive_plots.py

```diff
@@ -0,0 +1,160 @@
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import json
+import logging
+import os
+import shutil
+import time
+from pathlib import Path
+from typing import Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import plotly.express as px
+from IPython import get_ipython
+from IPython.display import clear_output, display
+from syne_tune.constants import SYNE_TUNE_DEFAULT_FOLDER
+from syne_tune.experiments import load_experiment
+
+
+def plot_tuner_results(tuner_name: str,
+                      experiment_name: str,
+                      return_df: bool = False,
+                      refresh: bool = False,
+                      refresh_rate: int = 60,
+                      height: Optional[int] = None,
+                      width: Optional[int] = None,
+                      max_rows: int = 100,
+                      max_cols: int = 100,
+                      display_dataframe: bool = True,
+                      display_metadata: bool = False,
+                      dimensions_max_cardinality: int = 50,
+                      save_figure: bool = False,
+    ) -> Optional[pd.DataFrame]:
+
+    """
+    This assumes the SageMaker output path of your hpo estimator is of the format:
+    output_path = f's3://{bucket}/syne-tune/{hpo_experiment_name}'
+
+    :param tuner_name: name of a tuning experiment previously run
+    :param experiment_name: name of a single experiment to load
+    :param return_df: whether to return a results dataframe
+    :param refresh: whether to refresh plotly figure every refresh_rate seconds
+    :param refresh_rate: refresh rate in seconds
+    :param height: height of the plotly figure
+    :param width: width of the plotly figure
+    :param max_rows: max rows to display in the displayed dataframe
+    :param max_cols: max columns to display in the displayed dataframe
+    :param display_dataframe: whether to display the dataframe
+    :param display_metadata: whether to display metadata
+    :param dimensions_max_cardinality: columns with more than this number of unique values are excluded
+    :param save_figure: whether to save the figure
+    :return: Optional[DataFrame]
+    """
+
+    if not get_ipython() is not None:
+        raise Exception("This function can only be used in a notebook")
+
+    pd.options.display.max_columns = max_rows
+    pd.options.display.max_rows = max_cols
+
+    while True:
+        ready_to_display = False
+
+        # Clear cached data
+        clear_output(wait=True)
+        syne_local_path = os.path.join(str(Path.home()), SYNE_TUNE_DEFAULT_FOLDER)
+        if os.path.exists(syne_local_path):
+            logging.info('clearing local syne tune cache...')
+            shutil.rmtree(syne_local_path)
```

Comment:
This will wipe out all experiments stored locally by default, could you remove it or use another temporary directory?

Reply:
How's this update? I changed it to just clear the single job so that it can continue to overwrite and retrieve live updates

Reply:
Ran `dev-setup.sh` committing